libclamav/ole2_extract.c

```diff
@@ -2186,9 +2191,19 @@ static bool verify_key_aes(const encryption_key_t *const key, encryption_verifie
     uint8_t decrypted[AES_VERIFIER_HASH_LEN];
     uint32_t tmp = 0;
 
-    if (!aes_128ecb_decrypt(verifier->encrypted_verifier, sizeof(verifier->encrypted_verifier),
-                            decrypted, key)) {
-        goto done;
+    switch (key->key_length_bits){
+        case SE_HEADER_EI_AES128_KEYSIZE:
+            if (!aes_128ecb_decrypt(verifier->encrypted_verifier, sizeof(verifier->encrypted_verifier),
+                                    decrypted, key)) {
+                goto done;
+            }
+            break;
+        case SE_HEADER_EI_AES192_KEYSIZE:
+            /*fallthrough*/
```

Comment:
I think we try to format these comments like this:
`/* fall-through */`
instead of like:
`/*fallthrough*/`

In searching the code, I see the former occurs in most of our newer code, and `/*fallthrough*/` is only done in one place in the PDF parser.  I suppose gcc must accept either one for silencing the fallthrough warnings, because I haven't seen it complain about them. 🤷

Reply:
I was just experimenting with clang-12, and it doesn't seem to accept any of them.  I'll change it to fall-through